### PR TITLE
Update logging-agent, zmon-worker and zmon-agent

### PR DESCRIPTION
**Pull request by mohabusama**
Changes:

**logging-agent**
- Updated scalyr-agent with ``copy_from_start`` option.
- Persist scalyr-agent ``checkpoints`` file to avoid log duplication.
- Drop dependency on ``ConfigMap`` for initial scalyr config and use ``initContainer`` instead.
- Update log-watcher manifest with modified volume types (``hostPath`` iso ``emptyDir``).
- Update log-watcher with adjusted rate limits for Journald and more debug info.

https://github.com/zalando-incubator/kubernetes-log-watcher/issues/28
https://github.com/zalando-incubator/kubernetes-log-watcher/issues/30
https://github.com/zalando-incubator/kubernetes-log-watcher/issues/31
https://github.com/scalyr/scalyr-agent-2/issues/48

**zmon-worker**
- Add support kubernetes wrapper

https://github.com/zalando-zmon/zmon-worker/issues/203

**zmon-agent**
- Add Cluster Alias to entities.

https://github.com/zalando-zmon/zmon-agent-core/issues/2

### CODE REVIEW

File: cluster/manifests/zmon-agent/deployment.yaml [hunk 1]
```diff
@@ -45,6 +45,8 @@ spec:
 
             - name: ZMON_AGENT_KUBERNETES_CLUSTER_ID
               value: "{{ .ID }}"
+            - name: ZMON_AGENT_KUBERNETES_CLUSTER_ALIAS
+              value: "{{ .Alias }}"
```
Comment (mikkeloscar) [inline cluster/manifests/zmon-agent/deployment.yaml:49]: woo :D

**Comment by mikkeloscar** [comment]
:+1:

File: cluster/manifests/zmon-agent/deployment.yaml (hunk 1 above)
Comment (Raffo) [reply cluster/manifests/zmon-agent/deployment.yaml:49]: He wanted the Alias, I gave him the alias xD

**Comment by hjacobs** [comment]
:+1:

**Comment by hjacobs** [comment]
@mohabusama it still has the "do-not-merge" label? Should we wait for something from you before merging?

**Comment by mohabusama** [comment]
@hjacobs I am looking into the initContainer issue.

**Comment by mohabusama** [comment]
@hjacobs I have done a test and found that the initial change would duplicate the logs of existing applications, since the ``checkpoint`` and ``scalyr-logs`` symlinks are not persisted on the host with the current manifests. Was thinking if doing the transition in two steps:

1. Update current manifests to use ``hostPath`` for scalyr-logs and checkpoint (roll-out to live clusters)
2. Update live clusters with this PR

or is the duplication issue negligible?

**Comment by hjacobs** [comment]
@mohabusama the duplication might be max. 50 MB right? (Docker log rotation is set to 50MB)

**Comment by mohabusama** [comment]
@hjacobs I think less, because Scalyr agent puts a limit of 5MB buffer.

https://github.com/scalyr/scalyr-agent-2/issues/48#issuecomment-285803497

**Comment by Raffo** [comment]
If the duplication is only once, while updating the clusters, I think we are good to go and it's okay to merge. I don't see it as a problem if we have a negligible duplication once.

**Comment by hjacobs** [comment]
:+1:

**Comment by Raffo** [comment]
👍

**Pull request merged**